### DISCUSSION TITLE
How to implement a 24 hour dial?

### DISCUSSION THREAD

**Discussion by ashishbhate** [zepp-health/zeppos-forum#391] [Q&A] [answered]
Hi,

Watches usually have a 12 hour dial, i.e. the hour hand (pointer) completes one rotation every 12 hours (so two rotations for each day).

I want to make a watchface that implements a 24 hour dial, i.e the hour hand (pointer) only makes one rotation per day. Is this possible? 

I was looking at the watch face API
https://docs.zepp.com/docs/watchface/api/hmUI/widget/TIME_POINTER/
This API only lets us define the pointer images, but the rotation of the image based on the current time is handled by Zepp OS, which only implements a 12 hour dial.

I think maybe I can use the [IMG_POINTER](https://docs.zepp.com/docs/watchface/api/hmUI/widget/IMG_POINTER/) widget to calculate the image rotation myself, but I'm not sure how to hook this up with the current time so that the widget updates in real time (except maybe deleting and recreating the widget every second, which seems inefficient)

**Comment by silver-zepp** [comment, marked as answer]
@ashishbhate You can use [setProperty](https://docs.zepp.com/docs/reference/device-app-api/newAPI/ui/setProperty) modifier. No need to recreate the widget. Use [Time](https://docs.zepp.com/docs/reference/device-app-api/newAPI/sensor/Time) library to get the current time, then map it to degrees (360/24).

**Reply by ashishbhate** [reply]
Thanks @silver-zepp. A follow-up question - 
Do I need to setup a `setInterval` to the redraw the watch face every second (with my angle calculations etc), or does Zepp OS automatically call the page's build method every second for a watchface?

**Reply by ashishbhate** [reply]
Ah, never mind. Looking at the sample watchface code [1] looks like I have create a timer [2] to trigger updates to the UI 

[1] https://github.com/zepp-health/zeppos-samples/blob/main/watchface/1.0/timer/watchface/round/index.js#L418
[2] https://docs.zepp.com/docs/watchface/api/timer/createTimer/
